### DISCUSSION TITLE
`fold`

### DISCUSSION THREAD

**Discussion by mozesa** [louthy/language-ext#1092] [Q&A] [answered]
Hello,

I have an initial state as `ConnectionStatus.Unknown`, with which I want to call an `Aff<RT, ConnectionStatus, ConnectionStatus>` then call it again, but now using the return value from the previous run of `Aff<RT, ConnectionStatus, ConnectionStatus>` and so forth (recursively).

Some `Scheduling` would be awesome between two subsequent call of `Aff<RT, ConnectionStatus, ConnectionStatus>`.

I haven't managed to figure it out yet.

Thanks for any tips in advance.

**Comment by louthy** [comment, marked as answer]
Take a look at the [`FoldUntil` and `FoldWhile` extensions](https://github.com/louthy/language-ext/blob/main/LanguageExt.Core/Effects/Aff/Extensions/Aff.Extensions.Fold.cs)
